sympy/stats/drv.py

```diff
@@ -199,6 +226,21 @@ def conditional_space(self, condition):
         domain = ConditionalDiscreteDomain(self.domain, condition)
         return DiscretePSpace(domain, density)
 
+    def compute_density(self, expr, **kwargs):
+        z = Dummy('z', real=True, finite=True)
+        return Lambda(z, self.integrate(KroneckerDelta(expr, z), **kwargs))
```

Comment:
We definitely need to rename `integrate` for two reasons:
1. it's not an integration in this case (rather a summation),
2. it's confusing as a name. People reading the code may mistakenly think it's a call to the integrals module.